test/test_model.py

```diff
@@ -54,8 +54,8 @@ def test_custom_model():
     pls = PLSRegression(n_components=4)
     pls.fit(X_train, Y_train)
     model1 = Model(names=["y1", "y2", "y3"])
-    model1.eval = lambda x: pd.DataFrame(
-        pls.predict(x.values), columns=model1.names, index=x.index
+    model1.__call__ = lambda df: pd.DataFrame(
+        pls.predict(df.to_numpy()), columns=model1.names, index=df.index
```

Comment:
This will have to be changed if the individual algorithms are going to also be altered. 
Or, said otherwise: sklearn likes to use nparrays as input instead of pandas dataframes...